tests/test_tools.py

```
@@ -0,0 +1,310 @@
+from functools import wraps
+from applefoundationmodels import Client
+
+
+class ToolTestHarness:
+    """Helper for testing tool calling with less boilerplate."""
+
+    def __init__(self, session):
+        self.session = session
+        self.calls = []
+
+    def register_tool(self, description: str = None):
```

Ruff (0.14.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)